File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        long long i=0;
        long long currSum=0;
        long long maxSum=INT_MIN;
        for(auto num:nums) {
            currSum+=num;
             maxSum = max(currSum,maxSum);

            if(currSum<0) {
                currSum=0;
                continue;
            }
        }
        return  maxSum;
        
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        sumOfSubArray = -1e7
        tempSumOfSubArray = 0
        lPointer=0
        rPointer=0
        while (rPointer<len(nums)):
            tempSumOfSubArray +=nums[rPointer]
            sumOfSubArray = max(tempSumOfSubArray,sumOfSubArray)
            if tempSumOfSubArray<0:
                lPointer=rPointer+1
                tempSumOfSubArray=0
            rPointer+=1
        return sumOfSubArray

        